### PR TITLE
Add audit events for headless login

### PR DESCRIPTION
**Pull request by AntonAM**
This PR adds login events for 

* User initiates headless login
* User approves/denies headless login request

Changelog: Add audit events for headless login.

Closes https://github.com/gravitational/teleport/pull/33297

### CODE REVIEW

File: lib/auth/methods.go [hunk 1]
```diff
@@ -438,6 +438,8 @@ func (s *Server) authenticateHeadless(ctx context.Context, req AuthenticateUserR
 		return nil, trace.Wrap(err)
 	}
 
+	emitHeadlessLoginEvent(ctx, events.UserHeadlessLoginRequestedCode, s.emitter, ha, nil)
```
Comment (AntonAM) [inline lib/auth/methods.go:441]: @Joerger I've added events as per RFD105, do you think it's ok to log this one? I think it's helpful, but not sure about the DoS concerns you discussed in the RFD.

**Review by Joerger — APPROVED** [review]

File: lib/auth/methods.go (hunk 1 above)
Comment (Joerger) [reply lib/auth/methods.go:441]: I think it is ok, the endpoint is rate limited, and it's better to log if someone is still trying to DOS.

File: web/packages/teleport/src/services/audit/makeEvent.ts [hunk 2]
```diff
@@ -699,6 +699,28 @@ export const formatters: Formatters = {
...
+    type: 'user.login',
+    desc: 'Headless Login Approved',
+    format: ({ user }) =>
+      `User [${user}] successfully approved headless login request`,
+  },
+  [eventCodes.USER_HEADLESS_LOGIN_APPROVEDFAILURE]: {
+    type: 'user.login',
+    desc: 'Headless Login Failed',
+    format: ({ user }) =>
+      `User [${user}] tried approve headless login request, but got an error`,
```
Comment (gzdunek) [inline web/packages/teleport/src/services/audit/makeEvent.ts:717]:
```suggestion
      `User [${user}] tried to approve headless login request, but got an error`,
```


Also, is it possible to print the error? I think we do it for the other failure events.

**Review by gzdunek — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/services/audit/makeEvent.ts (hunk 2 above)
Comment (AntonAM) [reply web/packages/teleport/src/services/audit/makeEvent.ts]: Added printing out of the error.

**Pull request merged**